fv_be/app/models/dictionary.py

```diff
@@ -0,0 +1,153 @@
+        return _("Translation in %(language)s: %(translation)s.") % {
+            "language": self.language,
+            "translation": self.translation,
+        }
+
+
+class AlternateSpelling(BaseModel):
+    """Model for alternate spellings associated to each dictionary entry."""
+
+    # from fv:alternate_spelling, fv-word:alternate_spellings, fv-phrase:alternate_spellings
+    text = models.CharField(max_length=200)
+    dictionary_entry = models.ForeignKey(
+        "DictionaryEntry", on_delete=models.CASCADE, related_name="alternate_spellings"
+    )
+
+    def __str__(self):
+        return self.text
+
+
+class Pronunciation(BaseModel):
+    """Model for pronunciations associated to each dictionary entry."""
+
+    # from fv-word:pronunciation
+    text = models.CharField(max_length=200)
+    dictionary_entry = models.ForeignKey(
+        "DictionaryEntry", on_delete=models.CASCADE, related_name="pronunciations"
+    )
+
+    def __str__(self):
+        return self.text
+
+
+class DictionaryEntry(BaseModel):
+    """Model for dictionary entries"""
+
+    class TypeOfDictionaryEntry(models.TextChoices):
+        # Choices for Type
+        WORD = "WORD", _("Word")
+        PHRASE = "PHRASE", _("Phrase")
+
+    # Fields
+    # from dc:title
+    title = models.CharField(max_length=200)
+    type = models.CharField(
+        max_length=6,
+        choices=TypeOfDictionaryEntry.choices,
+        default=TypeOfDictionaryEntry.WORD,
+    )
+    # todo: Link to site model when available
+    # may be inherited from an abstract base class or mixin later
+    # site = models.ForeignKey(
+    #     'Site',
+    #     on_delete=models.CASCADE
+    # )
+    # from fv-word:categories, fv-phrase:phrase_books
+    categories = models.ForeignKey(
+        Category,
+        blank=True,
+        null=True,
+        on_delete=models.SET_NULL,
+        related_name="dictionary_entries",
+    )
+    # from fv:custom_order
+    custom_order = models.CharField(max_length=200, blank=True)
+    # from fv-word:available_in_games, fvaudience:games
+    exclude_from_games = models.BooleanField(default=False)
+    # from fvaudience:children fv:available_in_childrens_archive
+    # exclude_from_kids can be a shared mixin for dictionary_entries, songs, stories and media
+    exclude_from_kids = models.BooleanField(default=False)
+    batch_id = models.CharField(max_length=255, blank=True)
+    # from fv:related_assets, fv-word:related_phrases
+    related_dictionary_entries = models.ManyToManyField(
+        "self",
+        blank=True,
+        symmetrical=False,
+        through="DictionaryEntrySelfAssociation",
+        related_name="incoming_related_dictionary_entries",
+    )
+
+    class Meta:
+        verbose_name_plural = "DictionaryEntries"
+
+    def __str__(self):
+        return self.title
+
+
+class DictionaryEntrySelfAssociation(models.Model):
+    from_dictionary_entry = models.ForeignKey(DictionaryEntry, on_delete=models.CASCADE)
```

Comment:
Should we cascade on delete here? I'd assume if a related entry is deleted that we would just null the relation.

Reply:
Agreed. Good catch :)

Reply:
Yes. We would not want the related entry to be delete in case one entry is deleted. I'll try to explain my understanding of this but please correct me if I'm wrong.

In this case, this DictionaryEntrySelfAssociation model is an intermediary ("through" in django terms) table to store the many to many relation. So what the CASCADE implies is, if word1 and word2 are related, and one word gets deleted, this just deletes the relation b/w those words, since there is no point in keeping a relation which has null in either 'from' or 'to' in the association table.